sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractRulesDefinition.java

```diff
@@ -81,10 +87,15 @@ private void setupHotspotRules(Collection<NewRule> rules) {
       .collect(Collectors.toSet());
   }
 
-  private static void updateSecurityStandards(NewRule rule, RuleMetadata ruleMetadata) {
-    for (String s : ruleMetadata.securityStandards.OWASP) {
+  private void updateSecurityStandards(NewRule rule, RuleMetadata ruleMetadata) {
+    for (String s : ruleMetadata.securityStandards.OWASP_2017) {
       rule.addOwaspTop10(RulesDefinition.OwaspTop10.valueOf(s));
     }
+    if (isOwaspByVersionSupported) {
+      for (String s : ruleMetadata.securityStandards.OWASP_2021) {
+        rule.addOwaspTop10(RulesDefinition.OwaspTop10Version.Y2021, RulesDefinition.OwaspTop10.valueOf(s));
+      }
```

Comment:
@pavel-mikula-sonarsource I don't think it's worth it. First we don't know when next version of OWASP will be released. Second, we still need to call previous API (without enum first argument) to support older versions of sonar-plugin-api. So IMHO it will be even a bigger mess with such "low-maintainance" optimization.